### PR TITLE
Trx cleanup

### PR DESCRIPTION
**Pull request by roytman**
remove unused methods
move etcd event print-related methods to database.go
allow combinations of "Select" and data unmodified operations
extend `TestTransactComment` to check the comment data
add utittest with combination of `Select` and `Comment`

### CODE REVIEW

File: pkg/ovsdb/database.go [hunk 1]
```diff
@@ -209,6 +212,87 @@ func (con *DatabaseEtcd) CreateMonitor(dbName string, handler *Handler, log logr
...
+	Key   common.Key
+	Value map[string]interface{}
+}
+
+func NewKeyValue(etcdKV *mvccpb.KeyValue) (*KeyValue, error) {
+	kv := new(KeyValue)
+
+	/* key */
+	key, err := common.ParseKey(string(etcdKV.Key))
+	if err != nil {
```
Comment (msherif1234) [inline pkg/ovsdb/database.go:279]: r u sure if no error that key ptr not nil or better to make sure key != nil before reference it below ?

File: pkg/ovsdb/database.go [hunk 2]
```diff
@@ -209,6 +212,87 @@ func (con *DatabaseEtcd) CreateMonitor(dbName string, handler *Handler, log logr
...
+	err = json.Unmarshal(etcdKV.Value, &kv.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	return kv, nil
+}
+
+func (kv *KeyValue) Dump() {
+	fmt.Printf("%s --> %v\n", kv.Key, kv.Value)
```
Comment (msherif1234) [inline pkg/ovsdb/database.go:293]: shouldn't we switch to logger pkg instead of fmt ?

File: pkg/ovsdb/transact.go [hunk 3]
```diff
@@ -591,17 +494,28 @@ func (txn *Transaction) AddSchema(databaseSchema *libovsdb.DatabaseSchema) {
...
+Loop:
 	for _, ovsOp := range txn.request.Operations {
-		if ovsOp.Op == OP_SELECT {
+		switch ovsOp.Op {
+		case OP_SELECT:
 			hasSelect = true
-		} else {
-			hasOther = true
+			if hasSelect && hasWrite {
+				break Loop
```
Comment (msherif1234) [inline pkg/ovsdb/transact.go:506]: hmm if u already set hasSelect to true why u are checking it and same below for hasWrite ?

File: pkg/ovsdb/database.go (hunk 1 above)
Comment (roytman) [reply pkg/ovsdb/database.go:279]: I can add the check, but I doubt that key in an etcd event can be  nil

File: pkg/ovsdb/database.go (hunk 2 above)
Comment (roytman) [reply pkg/ovsdb/database.go:293]: I have a better idea: there is no usage of the Dump method, so I will remove it for now

File: pkg/ovsdb/transact.go (hunk 3 above)
Comment (roytman) [reply pkg/ovsdb/transact.go:506]: just wrote the same test for the both switch cases, but you're right, fixed.

File: pkg/ovsdb/database.go (hunk 1 above)
Comment (roytman) [reply pkg/ovsdb/database.go:285]: added the checks (just in case)

**Pull request merged**